carp.protocols.core/src/commonMain/kotlin/dk/cachet/carp/protocols/domain/deployment/UnexpectedMeasuresWarning.kt

```diff
@@ -0,0 +1,40 @@
+package dk.cachet.carp.protocols.domain.deployment
+
+import dk.cachet.carp.common.data.DataType
+import dk.cachet.carp.protocols.domain.StudyProtocol
+import dk.cachet.carp.protocols.domain.devices.AnyDeviceDescriptor
+import dk.cachet.carp.protocols.domain.tasks.measures.Measure
+
+
+/**
+ * Evaluates whether a [StudyProtocol] contains [Measure]s that are requested on a device for which the requested [DataType] is not expected.
+ *
+ * This is allowed, but requires the client implementation to have corresponding support to handle this unexpected [DataType].
+ */
+class UnexpectedMeasuresWarning internal constructor() : DeploymentWarning
+{
+    /**
+     * Holds an unexpected [measure] which at some point in the [StudyProtocol] may be sent to [device].
+     */
+    data class UnexpectedMeasure( val device: AnyDeviceDescriptor, val measure: Measure )
+
+    override val description: String =
+        "The study protocol contains measures that are requested on a device for which the requested data type is not expected." +
+        "This is allowed, but requires the client implementation to have corresponding support to handle this unexpected data type."
+
+
+    override fun isIssuePresent( protocol: StudyProtocol ): Boolean = getUnexpectedMeasures( protocol ).any()
```

Comment:
How and when does the client determine which measures it supports?

Reply:
This is even before the study protocol (or rather, the `MasterDeviceDeployment` subset) reaches the client. These are warnings triggered during the creation of a study protocol, i.e., on the study manager.